packages/reporter/src/controllers/TelepathyReporterController.ts

```diff
@@ -0,0 +1,95 @@
+import axios from "axios"
+import { hexToNumber, Chain } from "viem"
+import winston from "winston"
+import "dotenv/config"
+
+import lightClientContractABI from "../ABIs/TelepathyContractABI.json"
+import adapterContractABI from "../ABIs/TelepathyAdapterABI.json"
+import Multiclient from "../MultiClient"
+import { ControllerConfig } from "../types/index"
+
+class TelepathyReporterController {
+  sourceChain: Chain
+  destinationChains: Chain[]
+  name: string = "telepathy"
+  logger: winston.Logger
+  multiClient: Multiclient
+  adapterAddresses: { [chainName: string]: `0x${string}` }
+  lightClientAddresses: { [chainName: string]: `0x${string}` }
+  proofURL: string
+  queryBlockLength: string
+  blockBuffer: string
+
+  constructor(configs: ControllerConfig) {
+    this.sourceChain = configs.sourceChain
+    this.destinationChains = configs.destinationChains
+    this.logger = configs.logger
+    this.multiClient = configs.multiClient
+    this.adapterAddresses = configs.adapterAddresses
+    this.lightClientAddresses = configs.data.lightClientAddresses
+    this.proofURL = configs.data.proofURL
+    this.queryBlockLength = configs.data.queryBlockLength
+    this.blockBuffer = configs.data.blockBuffer
+  }
+  async onBlocks(blockNumbers: string[]) {
```

Comment:
Why `blockNumbers` is not used? Why `blockNumbers` is `string[]`? Why do you call `getBlockNumber` here? Can you explain it to me as i don't get it?

Shouldn't be work more or less like this?

```js
....

const fromBlock =  blockNumbers[0]
const toBlock = blockNumbers[blockNumbers.length - 1]

const logs = await client.getContractEvents({
  address: this.lightClientAddresses[chain.name.toLocaleLowerCase()] as `0x${string}`,
  abi: lightClientContractABI,
  eventName: "HeadUpdate",
  fromBlock,
  toBlock
})
        
......
```

Reply:
For Telepathy, we are fetching blocks from destination chain on Light Client contract, like a pull based. While the other controllers use push based where they query block from source chain. 
To conform with the other reporters in `BlockListener.ts` `_fetchBlocks` function, I put this as the parameter in TelepathyReporterController.
```
   await Promise.all(this.controllers.map((_controller: any) => _controller.onBlocks(blocks)))
```

Another difference is that the `queryBlockLength` don't need to comply with the 256 recent blocks limitation because we are fetching the event of these blocks from the Light Client contract.

Reply:
you are right! i forgot it!